### PR TITLE
Add HTML report specification.

### PR DESCRIPTION
**Pull request by ming13**

### CODE REVIEW

**Comment by artem-zinnatullin** [comment]
Please rename file to `HTML_REPORT.md`

File: spec/REPORT.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+This page should be skipped if sharding was used to run the test suite.
+
+Contains a list of items with following properties:
+
+* device name;
+* test suite success tests count;
+* test suite failure tests count;
+* test suite ignored tests count.
+
```
Comment (artem-zinnatullin) [inline spec/REPORT.md]: test suite duration

File: spec/REPORT.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+## Test Suite
+
+Contains following blocks from top to bottom.
+
+* Summary with following properties:
+    * success tests count;
+    * failure tests count;
+    * ignored tests count;
+    * duration.
+* Search input.
```
Comment (artem-zinnatullin) [inline spec/REPORT.md]: (for now simple text search)

File: spec/REPORT.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    * class name;
+    * name;
+    * device name;
+    * duration.
+* Properties:
+    * list of key-value pairs.
+* Files:
+    * list of file links.
+* Screenshots:
+    * list of screenshot images.
```
Comment (artem-zinnatullin) [inline spec/REPORT.md]: grid

**Comment by artem-zinnatullin** [comment]
Please change commit message to PR title (`Report` is an action and by looking at old PR title I thought you made some changes to composer behavior…)

File: spec/REPORT.md (hunk 3 above)
Comment (ming13) [inline spec/REPORT.md]: Welcome to the Grid!

**Comment by artem-zinnatullin** [comment]
Linking with #11.

**Comment by artem-zinnatullin** [comment]
cc @oguseynov please review

**Comment by artem-zinnatullin** [comment]
cc @squibms please review as well

File: spec/REPORT.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
+# Terms
+
+* Test is a single functionality check.
+    * Properties:
+        * name;
+        * package name;
```
Comment (artem-zinnatullin) [inline spec/REPORT.md]:
```
* package name;
* class name;
* name;
```
?

File: spec/REPORT.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        * class name;
+        * duration;
```
Comment (artem-zinnatullin) [inline spec/REPORT.md]: durationNanos

File: spec/REPORT.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        * log;
```
Comment (artem-zinnatullin) [inline spec/REPORT.md]: `pathToLog`?

File: spec/REPORT.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        * device name;
```
Comment (artem-zinnatullin) [inline spec/REPORT.md]: adbDevice?

Comment (artem-zinnatullin) [inline spec/REPORT.md]: It's more abstract thing than device and combines both emulators and real devices

File: spec/REPORT.md (hunk 5 above)
Comment (artem-zinnatullin) [inline spec/REPORT.md]:
```diff
+ completion status: passed | failed | ignored;
```

File: spec/REPORT.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+        * list of files;
+        * list of screenshots.
+* Test suite is a combination of all tests available.
+    * Properties:
+        * list of tests;
+        * success tests count;
+        * failure tests count;
+        * ignored tests count;
+        * log;
+        * duration.
```
Comment (artem-zinnatullin) [inline spec/REPORT.md]:
```diff
+ list of devices;
```

File: spec/REPORT.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+        * list of properties — property is a key-value pair;
+        * list of files;
+        * list of screenshots.
+* Test suite is a combination of all tests available.
+    * Properties:
+        * list of tests;
+        * success tests count;
+        * failure tests count;
+        * ignored tests count;
+        * log;
```
Comment (artem-zinnatullin) [inline spec/REPORT.md]: log per each device

Comment (artem-zinnatullin) [inline spec/REPORT.md]: logcat
+ instrumentation output

File: spec/REPORT.md (hunk 6 above)
Comment (artem-zinnatullin) [inline spec/REPORT.md]: logcat

File: spec/REPORT.md (hunk 5 above)
Comment (artem-zinnatullin) [inline spec/REPORT.md]: + optional stacktrace

**Pull request merged**